scripts/integration/datadog-e2e-logs/compose.yaml

```diff
@@ -0,0 +1,98 @@
+version: '3'
+
+services:
+  # Generates random log data for consumption by the custom Agent check
+  log_generator:
+    image: docker.io/mingrammer/flog
+    command:
+      - "-f"
+      - "json"
+      - "-n"
+      - "1000"
+      - "-t"
+      - "log"
+      - "-o"
+      - "/var/log/a_custom.log"
+    volumes:
+      - log_path:/var/log/
+
+  # Receives log data from the `datadog-agent-vector` service and sends
+  # to the `fakeintake-agent-vector` service.
+  vector:
+    depends_on:
+      - fakeintake-agent-vector
+    build:
+      context: ${PWD}
+      dockerfile: tests/data/e2e/datadog/Dockerfile
```

Comment:
Rather than adding another Docker image here, could we reuse the `vector-test-runner-<version>:latest` image to save on build time?

In particular, we could append something like `cargo build --tests --lib --bins --features ${FEATURES}` to the end of the Dockerfile which would compile all of the assets for both integration and e2e tests. Integration tests will still execute `cargo nextest` and would require no additional compilation. E2E tests would use the `/target/debug/vector` binary and require no additional compilation either.

Reply:
I think this would run into the same problem that I encountered when attempting a prior solution than the existing one of having a separate Dockerfile/service for vector, which is that the `datadog-agent-vector` service needs to have a dependency on the Vector instance. The prior solution involved trying to add support to the `vdev` wiring up of the services etc. , such that a dependency "runner" could be used. That ended up becoming rather messy and complicated so I abandoned it for the current approach. Your suggestion here is simpler on the surface, than the vdev changes approach, but it's not clear to me how we would firstly know what the `<version>` would be to use dynamically and also that issue of needing to specify that as a dependency in the compose file.

Reply:
Thanks for the additional details on that!

I went down this path for a bit, but had issues getting the Dockerfile working. I stashed the changes and will come back to it.

Reply:
I think I finally sorted this out in https://github.com/vectordotdev/vector/pull/18538/commits/bedb3d3d62cbe84e3336cf3ba7d06cb132f522ca